proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+  string name = 1;
+
+  // The principal type.
+  Security.PrincipalType principal_type = 2;
+}
+
+// UpdateRoleMembershipRequest is the request to UpdateRoleMembership.
+message UpdateRoleMembershipRequest {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Create the role if it doesn't already exist.
+  // If the role is created in this way, the “add” list will be respected, but the “remove” list will be ignored.
+  optional bool create = 2;
+
+  // Members to assign to the role.
+  repeated RoleMembership add = 3;
+
+  // Members to remove from the role.
+  repeated RoleMembership remove = 4;
+}
+
+// UpdateRoleMembershipResponse is the response for UpdateRoleMembership.
+message UpdateRoleMembershipResponse {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Members assigned to the role.
+  repeated RoleMembership added = 2;
+
+  // Members removed from the role.
+  repeated RoleMembership removed = 3;
+}
+
+// ListUserRolesRequest is the request for ListUserRoles.
+message ListUserRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results to only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+  }
+
+  // Optional filter.
+  optional Filter filter = 1;
+
+  // Page size.
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+}
+
+// ListUserRolesResponse is the response for ListUserRoles.
+message ListUserRolesResponse {
+  // The roles in the system.
+  repeated Role roles = 1;
+}
+
+// ListRolesWithMembersRequest is the request for ListRolesWithMembers.
+message ListRolesWithMembersRequest {
+  // Filter options.
+  message Filter {
+    // Filter results only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+  }
+
+  // Optional filter.
+  optional Filter filter = 1;
```

Comment:
If no filter is provided we will list all roles with their members? I assume that each role may have thousands of members potentially = problematic for pagination? How do we offer pagination here? One item = one role with all members? Seems massive, is this done in the Redpanda Admin API the same way?

Reply:
The frontend expressed the need for a call that lists roles _and_ the role members.
I share the concern about not being able to control the size of individual list of members.
@rikimaru0345 can share more details and perhaps what's needed and how this would be used and how we can control response size.

Reply:
The frontend should not necessarily be the driver for the API. Keep in mind that we want to expose these APIs in the future. Also if one request may multiply many requests towards the backend, we need to be **very careful** - I want to avoid this as much as possible. Otherwise we run into a bunch of problems in regards to performance and effective rate limiting.

Reply:
I have removed `ListRolesWithMembers` rpc.